lib/eventq/eventq_aws/aws_eventq_client.rb

```diff
@@ -42,23 +42,25 @@ def raise_event(event_type, event, context = {})
         register_event(event_type)
 
         with_prepared_message(event_type, event, context) do |message|
-
+          # TODO: Would be nice to not have to make an API call to get the topic ARN.  Two calls for every publish.
+          # Should cache the ARN's into a mutex class variable collection.
+          topic_arn = topic_arn(event_type)
```

Comment:
I agree that `EventQ` should maintain an in memory cache list of arns to avoid making an extra API call to AWS for every event published to AWS. SNS publishing can already take between `60-300ms` depending on connection re-use (which is flakey at best in the v2 sdk) we want to avoid increasing publish time if we can.